src/aioauth/server.py

```diff
@@ -81,31 +156,53 @@ async def create_authorization_response(self, request: Request) -> Response:
 
         See Section 4.1.1: https://tools.ietf.org/html/rfc6749#section-4.1.1
         """
-        ResponseTypeClass: Union[
-            Type[ResponseTypeToken],
-            Type[ResponseTypeAuthorizationCode],
-            Type[ResponseTypeBase],
-        ] = self.response_type.get(request.query.response_type, ResponseTypeBase)
-        response_type = ResponseTypeClass(db=self.db)
-
-        response = await response_type.create_authorization_response(request)
-
-        response_dict = {
-            **response._asdict(),
-            "state": request.query.state,
-        }
-        query_params = (
-            response_dict
-            if request.query.response_type == ResponseType.TYPE_CODE
-            else {}
-        )
-        fragment = (
-            response_dict
-            if request.query.response_type == ResponseType.TYPE_TOKEN
-            else {}
-        )
+        self.validate_request(request, [RequestMethod.GET])
+
+        response_type_list = str_to_list(request.query.response_type)
+        response_type_classes: Set[AnyResponseTypeClass] = set()
+
+        responses = {}
+        fragment = {}
+        query = {}
 
-        location = build_uri(request.query.redirect_uri, query_params, fragment)
+        if not response_type_list:
+            # NOTE: In case of empty response_type, the validator of
+            # ResponseTypeBase class will raise InvalidRequestError.
+            raise InvalidRequestError(
+                request=request, description="Missing response_type parameter."
+            )
+
+        if request.query.state:
+            responses["state"] = request.query.state
+
+        for response_type in response_type_list:
+            ResponseTypeClass: Optional[AnyResponseTypeClass] = self.response_types.get(
```

Comment:
I usually almost never type-hint arguments personally- I use an explicit name to describe the variable. Any reason for this design choice here? No against or for, just curious.

Reply:
this actually interesting story :) since python do not exactly know what type will be in response_type_list, I specified that explicitly, because vscode autocompletion works well in this case. but I agree that this is not necessary.